pkg/inject/sidecar_test.go

```diff
@@ -616,3 +636,30 @@ func TestSidecarArgumentsOpenshiftTLS(t *testing.T) {
 	agentTags := agentTags(dep.Spec.Template.Spec.Containers[1].Args)
 	assert.Contains(t, agentTags, "container.name=only_container")
 }
+
+func TestEqualSidecar(t *testing.T) {
+	jaeger := v1.NewJaeger(types.NamespacedName{
+		Name:      "TestEqualSidecar",
+		Namespace: "test",
+	})
+
+	dep1 := dep(map[string]string{Annotation: jaeger.Name}, map[string]string{})
+	dep1 = Sidecar(jaeger, dep1)
+
+	dep1Equal := dep(map[string]string{Annotation: jaeger.Name}, map[string]string{})
+	dep1Equal = Sidecar(jaeger, dep1Equal)
+	assert.True(t, EqualSidecar(dep1, dep1Equal))
+
+	// Change flags.
+	jaeger.Spec.Agent.Options = v1.NewOptions(map[string]interface{}{
+		"--jaeger.tags": "changed-tag=newvalue",
+	})
+
+	dep2 := dep(map[string]string{Annotation: jaeger.Name}, map[string]string{})
+	dep2 = Sidecar(jaeger, dep2)
+	dep3 := dep(map[string]string{Annotation: jaeger.Name}, map[string]string{})
+
+	assert.False(t, EqualSidecar(dep1, dep2))
```

Comment:
nit: Can you move this above `dep3`, and then before `dep3` line have a comment about `// Compare with no agent` or something like that - at the moment the different criteria being tested are being interleaved.